Classes/Service/DemandService.php

```diff
@@ -45,7 +47,11 @@ public function getRestrictionsFromDemandProviders(
         ExpressionBuilder $expressionBuilder
     ): CompositeExpression
     {
-        // TODO: Set $demandArray to merged, returned values of $demandProviders
+        $demandArray = [];
+
+        foreach ($demandProviders as $demandProvider) {
+            $demandArray = $demandProvider->getDemand();
```

Comment:
You have to merge the arrays, not overwrite them.

```suggestion
            $demandArray = array_merge_recursive($demandArray, $demandProvider->getDemand());
```

PS! There is also a more sophisticated array merge method in the TYPO3 API that is sometimes useful when merging configurations: `ArrayUtility::mergeRecursiveWithOverrule()` https://api.typo3.org/10.4/class_t_y_p_o3_1_1_c_m_s_1_1_core_1_1_utility_1_1_array_utility.html#ab64f1500617cb3ef04ec2f3e58490b11